paddle/fluid/operators/controlflow/while_op.cc

```diff
@@ -87,14 +101,48 @@ class WhileOp : public framework::OperatorBase {
     bool is_test = Attr<bool>("is_test");
     auto &skip_vars = Attr<std::vector<std::string>>(kSkipEagerDeletionVars);
     VLOG(2) << GetSkipEagerDeletionVarsDebugString(skip_vars);
-
     auto ctx = executor.Prepare(*program, block->ID(), skip_vars);
     if (!is_test) {
       while (cond_data) {
         auto &current_scope = scope.NewScope();
         step_scopes->push_back(&current_scope);
+        // Each loop the value of LoDTensor in input would be saved sub_scope to
+        // ensure that the correct values can be used in the gradient
+        // calculation.
+        auto input_names = Inputs(kX);
```

Comment:
"auto" -> std::vector<std::string>, then you can use the auto next line.

Reply:
Thanks, done.